### PR TITLE
Div 8487 | API to request for OTP through MOSIP

### PR DESCRIPTION
**Pull request by samarthgr**

### CODE REVIEW

File: backend/registration_api/go.mod [hunk 1]
```diff
@@ -14,21 +14,15 @@ require (
...
 	github.com/imroc/req v0.3.0
 	github.com/jessevdk/go-flags v1.4.0
 	github.com/jinzhu/configor v1.2.1
-	github.com/onsi/ginkgo v1.15.0 // indirect
-	github.com/onsi/gomega v1.10.5 // indirect
+	github.com/lestrrat-go/jwx v1.2.24
 	github.com/pkg/errors v0.9.1
 	github.com/sirupsen/logrus v1.7.0
-	golang.org/x/net v0.0.0-20201202161906-c7110b5ffcbb
-	google.golang.org/protobuf v1.25.0 // indirect
```
Comment (egov-joy) [inline backend/registration_api/go.mod:31]: Are the removed libraries all unused?

File: backend/registration_api/pkg/services/mosip_service.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		IndividualIdType: individualIDType,
+		OtpChannel:       []string{"email"},
+	}
+
+	reqJson, err :=json.Marshal(requestBody)
+	if err != nil {
+		log.Errorf("Error occurred while trying to unmarshal the array of enrollments (%v)", err)
+		return err
+	}
+	log.Infof("requestBody before signature - %s", reqJson)
```
Comment (egov-joy) [inline backend/registration_api/pkg/services/mosip_service.go:51]: Production log levels might be set to info in some implementations. We should probably change these log statements to debug level.

File: docker-compose.yml [hunk 3]
```diff
@@ -180,6 +180,10 @@ services:
       ETCD_AUTH_ENABLED: $ETCD_AUTH_ENABLED
       ETCD_USERNAME: $ETCD_USERNAME
       ETCD_PASSWORD: $ETCD_PASSWORD
+      MOSIP_AUTH_TOKEN: $MOSIP_AUTH_TOKEN
```
Comment (egov-joy) [inline docker-compose.yml:183]: Please add the new env vars to .env.example as well.
MISP_LICENSE_KEY, PARTNER_ID and PARTNER_API_KEY will probably need a MOSIP_ prefix so they can be easily identified as MOSIP related configurations.

File: backend/registration_api/go.mod (hunk 1 above)
Comment (samarthgr) [reply backend/registration_api/go.mod:31]: This was removed after running `go mod tidy`

**Pull request merged**